Sources/StreamChat/Utils/Timers.swift

```diff
@@ -84,17 +84,17 @@ private class RepeatingTimer: RepeatingTimerControl {
     }
     
     private var state: State = .suspended
-    private let timer: DispatchSourceTimer
+    private var timer: DispatchSourceTimer?
     
     init(timeInterval: TimeInterval, queue: DispatchQueue, onFire: @escaping () -> Void) {
         timer = DispatchSource.makeTimerSource(queue: queue)
-        timer.schedule(deadline: .now() + .milliseconds(Int(timeInterval)), repeating: timeInterval, leeway: .seconds(1))
-        timer.setEventHandler(handler: onFire)
+        timer?.schedule(deadline: .now() + .milliseconds(Int(timeInterval)), repeating: timeInterval, leeway: .seconds(1))
+        timer?.setEventHandler(handler: onFire)
     }
     
     deinit {
-        timer.setEventHandler {}
-        timer.cancel()
+        timer?.setEventHandler {}
+        timer?.cancel()
```

Comment:
```suggestion
        timer?.setEventHandler {}
        time?.invalidate()
        timer?.cancel()
```
Don't we need to call invalidate here? I don't remember if this is still necessary or not 🤔

Reply:
We're leveraging `DispatchSourceTimer` internally.